Killing Plot Adaptation

zumbak04 opened the pull request:
Fixes #1141, closes #499.
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- **Restricted some normal killing plots:**
  - Only those who can drink wine and organize banquets (so not tribal, nomadic rulers, demons, undead, etc.), can be killed by the **Poisoned Wine** plot.
  - Only humanoids who can't fly and like to watch the sunrise (so not harpies, Necromantic (religion) characters, etc.) can be killed by the **Falling Accident** plot.
  - Only those who like to show in public and sit in a carriage (so not nomadic, tribal, Necromantic (religion) characters, centaurs, etc.) can be killed by the **Archer Ambush** plot.
  - Only those who like to show in public and enter forgotten taverns (so not nomadic, tribal, Necromantic (religion) characters, etc.) can be killed by the **Manure Explosion** plot.
  - Only humanoids who can't breathe underwater (so not murlocs, undead, etc.) can be killed by the **Bathtub Drowning** plot.
  - Only those who can't breathe underwater or fly (so not murlocs, undead, harpies, etc.) can be killed by the **Boating Accident** plot.
  - Only those who aren't immune to natural venom (so not demons, undead, etc.) be killed by the **Poisonous Snake** plot.
  - Only those who can't fly and like to watch the sunrise (so not harpies, Necromantic (religion), tribal, nomadic characters, etc.) can be killed by the **Balcony Plummet** plot.
  - Only those who can hunt (so not **Elunic** religion, etc.) can be killed by the **Hunting Accident** plot.
  - Only humanoids who can't fly and sit in a carriage (so not harpies, centaurs, etc.) can be killed by the **Сarriage Accident** plot.
- **Restricted some killing in hiding plots:**
  - Only those who can get sick (so not undead, demons, etc.), can't be killed by the **Poisoned Food** plot.
  - Only those who aren't immune to natural venom (so not demons, undead, etc.) be killed by the **Snake Charmer** plot.
  - Only humanoids can be killed by the **Bribed Nurse** plot.
- Reworked the **Disguised Highwaymen** plot. It can fire for children and incapable characters. Plus, instead of highwaymen, it may mention:
  - Dragon hunters if the plot target is a dragon (for the future, you can't plot against dragons right now).
  - Monster hunters if the plot target is a demon, undead, void being, worgen.
  - Witch hunters if the plot target practices Dark magic.
- Reworked the **Ninja** plot. It can fire for incapable characters.

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:
Play and check how plots work for different characters.

I added a game rule to **SLIGHTLY** increase the chances of killing plot attempts (of player only or both player and AI) to save your time. The rule is called **Killing Chances** and can be found at the end of the game rules.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/75096083-4c8d6d80-55b5-11ea-96e7-18b490120c6e.png)

</details>

Note: the game rule is here to save your time, it's not an object of testing.

Also, I recommend to write `yesman` in the console, so you can invite everyone in your plots.

bantheadmin commented on common/scripted_score_values/wc_scripted_score_values.txt:
```diff
@@ -212,4 +212,23 @@ stereotypical_traits_game_rule_score = {
+
+killing_speed_score = {
+	modifier = {
+		factor = 0.01
```
Is this the value that modifies the plot's speed? I could play around with it to see the difference more clearly.

zumbak04 replied:
Yes. As I know, this is the smallest factor that can be put. `0.001` won't work.

zumbak04 commented:
> Noticed that with the game rule active, plotting starts a little faster. However, the time between the first event (co-plotter's suggestion) to the next event (plot executes) seems fixed (about 27 game days) and doesn't seem to be affected by the game rule. From the few AI plots I've tested, I assume the same thing happens.

My bad, I rephrased it in the changelog. The game actually affects the chances of killing plot attempts.

> Also, a suggestion to group the game rule with the other mod rules on top of the list (right now it's at the bottom)

This game rule is for testing only... or fun... very weird way of fun... that's why it's there.

zumbak04 commented:
> Maybe add "Recommended" text on Off option for clarification if modifying isn't intended for normal gameplay, or make the text similar to the CFS rule.

@bantheadmin added a warning and mentioned the exact factor of increasing.

zumbak04 commented on localisation/00_zuwc_united_localisation.csv:
```diff
@@ -2902,11 +2902,11 @@ GetAsYouRise_try;As you try to rise;;;;;;;;;;;;;x
+RULE_KILLING_SPEED_OFF_DESC;The chances of the killing plot attempts are unchanged.;;;;;;;;;;;;;x
 RULE_KILLING_SPEED_PLAYER;Player;;;;;;;;;;;;;x
-RULE_KILLING_SPEED_PLAYER_DESC;Player's chances of the killing plot attempts are SLIGHTLY increased.;;;;;;;;;;;;;x
+RULE_KILLING_SPEED_PLAYER_DESC;Player's chances of the killing plot attempts are increased by �G100�!.\n�RThe rule is created for testing. Use at your own risk.�!;;;;;;;;;;;;;x
```
I doubt players know what MTTH is. It's not a gameplay thing.

MemeWizard commented:
8164 - can characters, who can't drink, celebrate with beer?

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/76703080-b9d97d80-66df-11ea-8709-af39daf91928.png)


</details>

arithon commented:
Who can't drink?

MemeWizard commented:
> Monster hunters if the plot target is a demon, undead, void being, worgen.

Can't confirm, still see highwaymen/bandits

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/76704046-56ebe480-66e7-11ea-90bd-beb9f0c0bf5c.png)

</details>

> Who can't drink?

Liches, for example.

> Dragon hunters if the plot target is a dragon.

Except you can't kill dragons, but ok.

MemeWizard commented on common/scripted_triggers/wc_scripted_plot_triggers.txt:
```diff
@@ -0,0 +1,76 @@
+	is_humanoid_trigger = yes
+	can_get_sick_trigger = yes
+}
+can_be_killed_by_balcony_plummet_trigger = {
```
Event speaks about castle balcony, so it can be weird for tribal/nomadic rulers

zumbak04 replied:
Added tribal, nomad limitations in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/1219/commits/0d514b9deee420a8667987c748a9ca7512663f8c.

MemeWizard commented on common/scripted_triggers/wc_scripted_plot_triggers.txt:
```diff
@@ -0,0 +1,76 @@
+	dark_public_religion_trigger = no
+}
+can_be_killed_by_manure_explosion_trigger = {
+	#Tribal/nomadic ruler touring and entering taverns... sounds weird... don't know why Paradox haven't disabled it
```
What about tribal monarchy?

zumbak04 replied:
Well, it says nothing about a carriage and most likely they have some kind of taverns, so why not? Allowed in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/1219/commits/0d514b9deee420a8667987c748a9ca7512663f8c.

zumbak04 commented:
> 8164 - can characters, who can't drink, celebrate with beer?

Fixed in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/1219/commits/37e30cca8e08b634ee9a1486c47c786d0fa93106.

> > Monster hunters if the plot target is a demon, undead, void being, worgen.
> 
> Can't confirm, still see highwaymen/bandits

Fixed in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/1219/commits/ad85535aaa4dec0f1b215224b0c28cf4b9b1b77c.

MemeWizard commented:
What about event option?

<details>
<summary>This one</summary>

![image](https://user-images.githubusercontent.com/46576860/76705914-c9af8c80-66f4-11ea-9b65-d8e20d3a9243.png)

</details>

<details>
<summary>definition</summary>

![image](https://user-images.githubusercontent.com/46576860/76705965-1d21da80-66f5-11ea-87be-9c3c385c7f20.png)

</details>

MemeWizard commented on common/scripted_triggers/wc_scripted_plot_triggers.txt:
```diff
@@ -0,0 +1,80 @@
+	is_humanoid_trigger = yes
+	can_breath_underwater_trigger = no
+}
+can_be_killed_by_boating_accident_trigger = {
```
> Whispers have reached my ears that [Root.PlotTarget.GetTitledName] is due to embark on a little boat trip. I have it on good authority that [Root.PlotTarget.GetSheHe], like many children, has not yet learned to swim. If the boat was to flounder... well, these things happen all the time, don't they?

Can't they just fly away?

zumbak04 replied:
Make sense. Added harpies limit in eef083a90c49655de63b680162b54e0209b47683.

MemeWizard commented on localisation/00_wc_vanilla_replacements.csv:
```diff
@@ -292,4 +292,19 @@ EVTDESC_MNM_20081_CHILD;My fellows[Root.GetInSociety] have managed to insert the
 stop_abduct_decision_desc;Call off my [Root.GetSocietyMembers] from abducting my current target.;;;;;;;;;;;;;x
-bloodthirsty_mass_sacrifice_decision;Hold Blood Rituals;;;;;;;;;;;;;x
\ No newline at end of file
+bloodthirsty_mass_sacrifice_decision;Hold Blood Rituals;;;;;;;;;;;;;xEVTDESC7880;[Root.PlotTarget.GetTitledName] will soon travel in a convoy on an isolated road. Perhaps if we gathered some mercenaries and disguised them as [Root.PlotTarget.GetHighwaymenType]... crime is on the rise, after all.;;;;;;;;;;;;;x
```
Broken merge?

zumbak04 replied:
Fixed in 32d056c13669eb6a19d1b03951787f589f46499b.

MemeWizard commented on localisation/00_wc_vanilla_replacements.csv:
```diff
@@ -292,4 +292,19 @@ EVTDESC_MNM_20081_CHILD;My fellows[Root.GetInSociety] have managed to insert the
+EVTDESC7882;My co-conspirator, [From.GetTitledName], has asked for gold to hire some mercenaries. They will ambush [Root.PlotTarget.GetTitledName] disguised as [Root.PlotTarget.GetHighwaymenType].;;;;;;;;;;;;;x
+EVTDESC7885;You are travelling in a convoy in the countryside of [Root.Location.GetName] when it comes to a sudden halt. The road has been blocked by a fallen tree, and the convoy leader explains that you will have to make a slight detour on an isolated smaller path.;;;;;;;;;;;;;x
+EVTDESC7886;After less than half a league on the small path, you are ambushed and surrounded by [Root.GetHighwaymenType]! They send your guards on their way, explaining that their interest lies with you alone. You offer the belongings you carry with you, but they merely laugh at this - it is your blood they want.;;;;;;;;;;;;;x
+EVTOPTA7887;Y-you're not [Root.GetHighwaymenType]!;;;;;;;;;;;;;x
```
> You offer the belongings you carry with you, but they merely laugh at this - it is your blood they want.

Wow, *witch hunters* want your blood? What a surprise.

zumbak04 replied:
Rewrote `EVTOPTA7887` in b840de76a319f1b39da43829bcbceb858ae14cd3.

MemeWizard commented on localisation/00_wc_vanilla_replacements.csv:
```diff
@@ -292,4 +292,19 @@ EVTDESC_MNM_20081_CHILD;My fellows[Root.GetInSociety] have managed to insert the
+EVTDESC7886;After less than half a league on the small path, you are ambushed and surrounded by [Root.GetHighwaymenType]! They send your guards on their way, explaining that their interest lies with you alone. You offer the belongings you carry with you, but they merely laugh at this - it is your blood they want.;;;;;;;;;;;;;x
+EVTOPTA7887;Y-you're not [Root.GetHighwaymenType]!;;;;;;;;;;;;;x
+EVTDESC7888;You force your way through the improvised barricade, narrowly avoiding an ambush by [Root.GetHighwaymenType]! They send an accurate volley of arrows after your convoy as it speeds away, missing it by mere inches.;;;;;;;;;;;;;x
+EVTOPTA7889;Too accurate for [Root.GetHighwaymenType]. Only mercenaries are so precise...;;;;;;;;;;;;;x
```
Too accurate for *dragon hunters*? Are you sure about that localization?

zumbak04 replied:
Rewrote `EVTOPTA7889` in 671c0df44125d19949b01f35deebef9cfe4d231e.

zumbak04 commented:
> What about event option?

Changed in cf2379822bf354e7318e779ec0b03a762dc363e1.

arithon commented:
@MemeWizard
> Who can't drink?
Liches, for example.

https://youtu.be/1fzXmJyolfY?t=42
don't see the problem :D

arithon commented:
> Only those who like to show in public and sit in a carriage (so not nomadic, tribal, Necromantic characters, etc.) can be killed by the Archer Ambush plot.

Why can't you kill a nomad (in the open) by archers?

zumbak04 commented:
> Why can't you kill a nomad (in the open) by archers?

It talks that you travel around your city in a carriage.

arithon commented:
> Only humanoids who can't fly and like to watch the sunrise (so not harpies, Necromantic characters, etc.) can be killed by the Falling Accident plot.

What about druids with talon form mastery?^^

arithon commented:
> Only those who like to show in public and enter forgotten taverns (so not nomadic, tribal, Necromantic characters, etc.) can be killed by the Manure Explosion plot.

Manure explosions might work in abandoned laboratories, crypts etc as well, I don't see a reason to not using manure on undeads...

zumbak04 commented:
> What about druids with talon form mastery?^^

Shiiit.

> Manure explosions might work in abandoned laboratories, crypts etc as well, I don't see a reason to not using manure on undeads...

You need to change too much localization. It talks about a tavern, innkeeper bring something to you and your comrades, etc.

arithon commented:
> Only humanoids who can't fly (so not harpies, etc.) can be killed by the Сarriage Accident plot.

Would exclude centaurs/similarly big races too...

zumbak04 commented:
> What about druids with talon form mastery?^^

Allowed talon druids in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/1219/commits/deea826d8c0177cea92c71e4db6ce723e6b50ae2.

zumbak04 commented:
> Would exclude centaurs/similarly big races too...

Excluded centaurs in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/commit/cd5092f5ff081cd8cc756c68a9705dd8f4719f50.
Didn't touch big races because big races may use big carriages.

bantheadmin commented:
Suggestion for flavor text on plot events: For infamous lore characters who have been bossfights at some point in WoW, instead of the usual ambush by highwaymen/x hunters, they are ambushed by a group of 5/10/25 "disguised as adventurers". 
If it's something doable I can go ahead and compile a list of characters.

zumbak04 commented:
> * (Undead) Nerubians targeted by venomous snake plots

I can't confirm, I see it's not possible because of the Undead trait.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/76879305-84739200-688f-11ea-847d-ef5d65dce591.png)

</details>

> * Undead Vrykul targeted by poisoned wine plots

I can't confirm. I see it's not possible because of the Undead trait.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/76879083-2cd52680-688f-11ea-9e0d-5af770ca5d4e.png)

</details>

> Other than that, I noticed a few races where 90% of plot events are poisoned wine plots. Either bad rng or target factors like court rank? Happened on Gnoll targets for example.

Should be fixed in 0e23a4e5194875caf2901ba14950cf7b5d794fef.

> Suggestion for flavor text on plot events: For infamous lore characters who have been bossfights at some point in WoW, instead of the usual ambush by highwaymen/x hunters, they are ambushed by a group of 5/10/25 "disguised as adventurers".
> If it's something doable I can go ahead and compile a list of characters.

It's doable.
I suppose Arthas is in this list, but what if Arthas follows the Holy Light?

bantheadmin commented:
> > * (Undead) Nerubians targeted by venomous snake plots

> > * Undead Vrykul targeted by poisoned wine plots
> 
> I can't confirm. I see it's not possible because of the Undead trait.

It was partly an oversight from my part, because when I tried with a landed undead Nerubian to use the "invite noble to court" decision, the courtier wasn't undead as I expected. Same thing might've happened with Vrykuls, wasn't able to reproduce it.

<details>
<summary>Click to expand</summary>

![ruler undead trait](https://user-images.githubusercontent.com/4930893/76885105-67d35c00-6887-11ea-81b5-493bc792a647.png)

![court noble no trait](https://user-images.githubusercontent.com/4930893/76885135-7457b480-6887-11ea-9522-a83e57b47653.png)

</details>

> Should be fixed in [0e23a4e](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/commit/0e23a4e5194875caf2901ba14950cf7b5d794fef).


> I suppose Arthas is in this list, but what if Arthas follows the Holy Light?

As I saw in your screenshots, there could be a "be Lich King" check for Arthas specifically. Or, the suggestion can be expanded to all evil targets for starters (undead/demon/void being) who hold a title, with 5/10/25/40 man adventurers spawning for Counts, Dukes, Kings, Emperors respectively.

zumbak04 commented:
> As I saw in your screenshots, there could be a "be Lich King" check for Arthas specifically. Or, the suggestion can be expanded to all evil targets for starters (undead/demon/void being) who hold a title, with 5/10/25/40 man adventurers spawning for Counts, Dukes, Kings, Emperors respectively.

@bantheadmin I understand it's a reference to WoW, BUT! I think it's too intrusive. I can add it, but I would make it really rare... like the Cthulhu event from the vanilla.

bantheadmin commented:
> @bantheadmin I understand it's a reference to WoW, BUT! I think it's too intrusive. I can add it, but I would make it really rare... like the Cthulhu event from the vanilla.

Very rare, yes sure! It could be almost like an easter egg, something delightful for WoW players when they see it pop up. If you plan on doing this, you can do the text in one of many ways, for example: either assassins as disguised adventurers or just misled adventurers following promises of rewards.
Also it sounds easier to do 5/10/25/40 on evil title holders, rather than specific lore characters. Up to you.

zumbak04 commented:
> Very rare, yes sure! It could be almost like an easter egg, something delightful for WoW players when they see it pop up. If you plan on doing this, you can do the text in one of many ways, for example: either assassins as disguised adventurers or just misled adventurers following promises of rewards.
> Also it sounds easier to do 5/10/25/40 on evil title holders, rather than specific lore characters. Up to you.

@bantheadmin added in 99fc7dabf3af545552bfe43c88d0aaaf773afe7c. Click on the link to open the comment, explained some stuff there.

zumbak04 commented:
> And you're right about the "oh those nasty x adventurers" line sounding a bit weird. Is there a clean way to do it without mentioning a number? If not, we can leave it as is for now and think a good replacement later.

We can, but it's too much work for a rare reference.

> P.S. How do I test on dragons? Can't plot against them unless I'm missing something.

You can't, it's for the future. I'll mention it in the changelog.

arithon commented:
I'd vote for changing number of mercenaries/adventurers (number) to (text).

bantheadmin commented:
> 
> 
> I'd vote for changing number of mercenaries/adventurers (number) to (text).

Text looks more consistent in plot events since you never see a number. But on the other hand it's an easter egg, and standing out isn't necessarily bad. I'm ok with either of options.

> I'd even delete number of adventurers in answer-option

Zumbak said it's too much work for this specifically, and I'd feel bad advocating for hardcoding or whatever's needed to do this.

zumbak04 commented:
> I'd vote for changing number of mercenaries/adventurers (number) to (text).

Done in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/1219/commits/989a90f52ff5e3dd6ec17f7d833709b4f998d3b9.

bantheadmin commented:
Also reporting a minor unrelated bug, which I found while playing in this PR.
Eredun (demon, Fel) AI was swaying and antagonizing other Eredun (demon, Fel). Noticed it when I switched to the antagonized and I got the Rival event to popup. Switched to the aggressor, couldn't reproduce antagonize, but I found out she was swaying someone as well.

<details>
<summary>Click to expand</summary>

![demon sway](https://user-images.githubusercontent.com/4930893/77257654-638bb200-6c7e-11ea-84f3-4eac83ec3d90.png)

</details>

zumbak04 commented:
@bantheadmin 

> Or the text could be customized to be more believable, for example poisoning Undead with holy water in the wine (is that a thing?). That would require much more work though and personally I don't think it's worth the hassle, especially if you add other races/traits/religion combination in the future.

It's too much work... like you need to rewrite the whole text. The archer ambush talks about you traveling in a carriage around your city. The manure explosion talks about you and your company going to a tavern for drinks and food.

> Also reporting a minor unrelated bug, which I found while playing in this PR.
> Eredun (demon, Fel) AI was swaying and antagonizing other Eredun (demon, Fel). Noticed it when I switched to the antagonized and I got the Rival event to popup. Switched to the aggressor, couldn't reproduce antagonize, but I found out she was swaying someone as well.

Interestingu. Can you report it [Here](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/issues/new/choose)?

I will address everything else soon.

zumbak04 commented:
@bantheadmin 

> On to the easter egg, being an accomplice shows highwaymen text on planning event. Not sure if fix is needed, end result is still ambushed by adventurers.

Good one. It should be monster hunters. Fixed in 6a9c13abda86bfe550bdf1a80232bd7e42d92059.

>   * Monster hunters if the plot target is a demon, undead, void being, worgen.

> Easter egg again, killing unlanded courtier with five adventurers text. Suggestion to change it to three adventurers for unlanded (as in a group quest in WoW) if that's easy to do. If not, ignore this.

Done in 32e75534c28ee1c52707c0db3a1fd26590f0244f.

MemeWizard commented:
`EVTOPTA8184` can be weird for characters who can't eat, like ancients/liches

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/77258900-b40b0b80-6c8e-11ea-8600-f374bac8f48a.png)

</details>

zumbak04 commented:
> `EVTOPTA8184` can be weird for characters who can't eat, like ancients/liches

Fixed in 109861e087d1b1bbb47d23fceca55c52cef21107.

BronzeBaron commented:
Got the wine poisoning event with a harpy target.
![image](https://user-images.githubusercontent.com/49666645/77807229-9ad2d800-705d-11ea-962b-cacd4c5749b6.png)
Maybe not so suitable for harpies? Harpys at banquets

zumbak04 commented:
> Maybe not so suitable for harpies? Harpy/Murloc and maybe Gnoll at banquets seems odd.

@BronzeBaron fair. I would say it looks weird for all kinds of savages, so I disabled it for tribal and nomadic characters in https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/1219/commits/5ff611f10c42b7c69d533efbd791360c9577c59b.

zumbak04 commented:
> Everything's looks in place, however as I mentioned before there's no variety for evil characters. Not necessarily a bad thing, just pointing it out in case plot text will be added in the future.

@bantheadmin I wouldn't prioritize it, though you may make [a suggestion](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/issues/new/choose).
